### PR TITLE
Screen2 setup

### PR DESCRIPTION
**Pull request by Obehi234**
Screen 2 functionality completed
userScreen
screen2

### CODE REVIEW

File: app/src/main/java/com/example/randomuserapplication/UserDetailsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package com.example.randomuserapplication
+
+import android.os.Bundle
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/randomuserapplication/UserDetailsFragment.kt]: remove boiler plate code

File: app/src/main/java/com/example/randomuserapplication/presentation/ui/UserListFragment.kt [hunk 2]
```diff
@@ -29,6 +31,13 @@ class UserListFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         setUpRV()
+        rvAdapter.onContactClickListener = {userId ->
+            userViewModel.setUserId(userId)
+            val action = UserListFragmentDirections.actionUserListFragmentToUserDetailsFragment()
+            findNavController().navigate(action)
+            Log.d("CHECK_ID", "USER ID is $userId")
+        }
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/randomuserapplication/presentation/ui/UserListFragment.kt:36]: remove logs

File: app/src/main/java/com/example/randomuserapplication/viewmodel/UserViewModel.kt [hunk 3]
```diff
@@ -35,4 +38,16 @@ class UserViewModel @Inject constructor(
         }
     }
 
+    fun setUserId(userId: String) {
+        _selectedUserId.value = userId
+    }
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/randomuserapplication/viewmodel/UserViewModel.kt]: Why do u have this function to setUserId?

Comment (Obehi234) [reply app/src/main/java/com/example/randomuserapplication/viewmodel/UserViewModel.kt]: The purpose of this function is to ensure that the userId is set to the accurate id of the user being clicked and to be accessed by screen2 via the viewpager in order to retrieve the appropriate user details

Comment (IfyUnoka4) [reply app/src/main/java/com/example/randomuserapplication/viewmodel/UserViewModel.kt]: you can do that without the extra function in the viewmodel, when navigating, pass the user id, get it in the details screen and use it to get UserDetails

Comment (Obehi234) [reply app/src/main/java/com/example/randomuserapplication/viewmodel/UserViewModel.kt]: like safe args?

Comment (IfyUnoka4) [reply app/src/main/java/com/example/randomuserapplication/viewmodel/UserViewModel.kt]: yes

Comment (Obehi234) [reply app/src/main/java/com/example/randomuserapplication/viewmodel/UserViewModel.kt]: I removed unnecessary function and used nav safe args instead.

File: app/src/main/java/com/example/randomuserapplication/UserDetailsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.example.randomuserapplication
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/randomuserapplication/UserDetailsFragment.kt:5]: this is not needed

File: app/src/main/java/com/example/randomuserapplication/adapter/UserAdapter.kt [hunk 5]
```diff
@@ -23,11 +24,18 @@ class UserAdapter : ListAdapter<UserEntity, UserAdapter.UserViewHolder>(UserDiff
...
+                ContactBtn.setOnClickListener {
+                    val userId = item.id.toString()
+                    onContactClickListener?.invoke(userId)
+                }
+
+
                 followBtn.setOnClickListener {
                     followBtn.text = "Following"
                     Toast.makeText(root.context, "Follow successful!", Toast.LENGTH_SHORT)
                         .show()
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/randomuserapplication/adapter/UserAdapter.kt:37]: extract texts

File: app/src/main/java/com/example/randomuserapplication/UserDetailsFragment.kt (hunk 4 above)
Comment (Obehi234) [reply app/src/main/java/com/example/randomuserapplication/UserDetailsFragment.kt:5]: I removed it o. Lemme show you a screenshot

File: app/src/main/java/com/example/randomuserapplication/adapter/UserAdapter.kt (hunk 5 above)
Comment (Obehi234) [reply app/src/main/java/com/example/randomuserapplication/adapter/UserAdapter.kt:37]: ok

File: app/src/main/java/com/example/randomuserapplication/UserDetailsFragment.kt (hunk 4 above)
Comment (Obehi234) [reply app/src/main/java/com/example/randomuserapplication/UserDetailsFragment.kt:5]: screen2a

Comment (Obehi234) [reply app/src/main/java/com/example/randomuserapplication/UserDetailsFragment.kt:5]: my working tree is up to date as well

**Review by IfyUnoka4 — APPROVED** [review]

**Pull request merged**